src/main/ApplicationUtils.cpp

```diff
@@ -56,7 +57,26 @@ runWithConfig(Config cfg)
                          << "(for testing only)";
         }
 
-        app->start();
+        if (cc)
+        {
+            Json::Value catchupInfo;
+            std::shared_ptr<HistoryArchive> archive;
+            auto const& ham = app->getHistoryArchiveManager();
+            archive = ham.selectRandomReadableHistoryArchive();
+            if (catchup(app, *cc, catchupInfo, archive) == 0)
+            {
+                // Pre-run catchup defers overlay startup.
+                app->getOverlayManager().start();
+            }
+            else
+            {
+                return 1;
```

Comment:
Should we return the error code returned by catchup? It returns 3 on failure from what I see and as a caller it can be nice to know which stage failed, Horizon may decide to retry differently if catchup failed for example

Reply:
Yep, good call. Fixed.